clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleSecurityGroupCachingAgent.groovy

```diff
@@ -210,6 +210,19 @@ class GoogleSecurityGroupCachingAgent extends AbstractGoogleCachingAgent impleme
                                                    "compute.firewalls.list", TAG_SCOPE, SCOPE_GLOBAL).items as List
 
         firewalls = (firewalls ?: []) + (hostFirewalls ?: [])
+        def newFirewalls = []
+        firewalls.each {
+          try {
+            if (GCEUtil.deriveProjectId(it.selfLink) != project) {
+              log.debug("${GCEUtil.deriveProjectId(it.selfLink)} does not equal ${project}")
+              newFirewalls.add(it)
+            }
+          } catch(Exception ex) {
+            log.debug(ex.getMessage())
+          }
+        }
+        log.debug("Old Size: ${firewalls.size()}")
+        log.debug("New Size: ${(firewalls.size() - newFirewalls.size())}")
```

Comment:
This is definitely not the write way to cache things properly, as right now the "New Size" is always 0, meaning none of them pass through the logic I wrote above. Which makes sense considering this is a similar approach I took in the function call itself. Any ideas where this should go instead? I assumed it would be around here since just about we get `firewalls = (firewalls ?: []) + (hostFirewalls ?: [])` which is the crux if my problems I believe.